src/mlpack/methods/reinforcement_learning/environment/continuous_mountain_car_V2.hpp

```diff
@@ -0,0 +1,224 @@
+  {
+   public:
+    /**
+     * Construct a state instance.
+     */
+    StateV2() : data(dimension, arma::fill::zeros)
+    { /* Nothing to do here. */ }
+
+    /**
+     * Construct a state based on the given data.
+     *
+     * @param data Data for the velocity and position.
+     */
+    StateV2(const arma::colvec& data): data(data)
+    { /* Nothing to do here. */ }
+
+    //! Modify the internal representation of the state.
+    arma::colvec& Data() { return data; }
+
+    //! Get the velocity.
+    double Velocity() const { return data[0]; }
+    //! Modify the velocity.
+    double& Velocity() { return data[0]; }
+
+    //! Get the position.
+    double Position() const { return data[1]; }
+    //! Modify the position.
+    double& Position() { return data[1]; }
+
+    //! Encode the state to a column vector.
+    const arma::colvec& Encode() const { return data; }
+
+    //! Dimension of the encoded state.
+    static constexpr size_t dimension = 2;
+
+   private:
+    //! Locally-stored velocity and position vector.
+    arma::colvec data;
+  };
+
+  /**
+   * Implementation of action of Continuous Mountain CarV2.
+   * In Continuous mountain car gain, the action represents the
+   * force to be applied. This value is bounded in range -1.0 to 1.0.
+   * Unlike the simple mountain car environment, where action space has a
+   * discrete value, continuous mountain car has continous action space
+   * value.
+   */
+  struct Action
+  {
+    double action[1];
+    // Storing degree of freedom
+    const int size = 1;
+  };
+
+  /**
+   * Construct a Continuous Mountain Car V2 instance using the given constant.
+   *
+   * @param positionMin Minimum legal position.
+   * @param positionMax Maximum legal position.
+   * @param positionGoal Final target position.
+   * @param velocityMin Minimum legal velocity.
+   * @param velocityMax Maximum legal velocity.
+   * @param velocityGoal Final target velocity.
+   * @param power Power generated by car.
+   */
+  ContinuousMountainCarV2(const double positionMin = -1.2,
+                        const double positionMax = 0.6,
+                        const double positionGoal = 0.45,
+                        const double velocityMin = -0.07,
+                        const double velocityMax = 0.07,
+                        const double velocityGoal = 0.06,
+                        const double power = 0.0015) :
+      positionMin(positionMin),
+      positionMax(positionMax),
+      positionGoal(positionGoal),
+      velocityMin(velocityMin),
+      velocityMax(velocityMax),
+      velocityGoal(velocityGoal),
+      power(power)
+  { /* Nothing to do here */ }
+
+  /**
+   * Dynamics of Continuous Mountain CarV2. Get reward and next state based
+   * on current state and current action.
+   *
+   * @param state The current state.
+   * @param action The current action.
+   * @param nextState The next state.
+   * @return reward, it's always -1.0.
+   */
+  double Sample(const State& state,
```

Comment:
You missed the one, I guess afterwards the code should build just fine.

Reply:
Looks like there are still some references to ``State``, perhaps a search and replace is helpful here.